tests/virtctl/usbredir.go

```diff
@@ -0,0 +1,271 @@
+
+	"kubevirt.io/kubevirt/pkg/libvmi"
+)
+
+// Capabilities from client side
+var helloMessageLocal = []byte{
+	0x00, 0x00, 0x00, 0x00, 0x44, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x75, 0x73, 0x62, 0x72,
+	0x65, 0x64, 0x69, 0x72, 0x20, 0x30, 0x2e, 0x31, 0x30, 0x2e, 0x30, 0x00, 0x00, 0x00, 0x00, 0x00,
+	0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+	0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+	0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0xff, 0x00, 0x00, 0x00,
+}
+
+// Expected capabilities from QEMU's usbredir
+var helloMessageRemote = []byte{
+	0x00, 0x00, 0x00, 0x00, 0x44, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x71, 0x65, 0x6d, 0x75,
+	0x20, 0x75, 0x73, 0x62, 0x2d, 0x72, 0x65, 0x64, 0x69, 0x72, 0x20, 0x67, 0x75, 0x65, 0x73, 0x74,
+	0x20, 0x35, 0x2e, 0x32, 0x2e, 0x30, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+	0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+	0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0xff, 0x00, 0x00, 0x00,
+}
+
+type ctxKeyType string
+
+const connectedKey ctxKeyType = "connected"
+
+var _ = Describe("[crit:medium][vendor:cnv-qe@redhat.com][level:component][sig-compute] USB Redirection", decorators.SigCompute, func() {
+	const (
+		enoughMemForSafeBiosEmulation = "32Mi"
+		vmiRunTimeout                 = 90
+	)
+
+	Describe("[crit:medium][vendor:cnv-qe@redhat.com][level:component] A VirtualMachineInstance with usbredir support", func() {
+		var vmi *v1.VirtualMachineInstance
+
+		BeforeEach(func() {
+			// A VMI for each test to have fresh stack on server side
+			vmi = libvmi.New(libvmi.WithResourceMemory(enoughMemForSafeBiosEmulation), withClientPassthrough())
+			vmi = libvmops.RunVMIAndExpectLaunch(vmi, vmiRunTimeout)
+		})
+
+		It("Should fail when limit is reached", func() {
+			type session struct {
+				cancel context.CancelFunc
+				err    chan error
+			}
+			var tests []session
+			for i := range v1.UsbClientPassthroughMaxNumberOf + 1 {
+				Eventually(func(g Gomega) bool {
+					cmd := newVirtctlCommand("usbredir",
+						"--namespace", vmi.ObjectMeta.Namespace,
+						"--no-launch", vmi.ObjectMeta.Name)
+					connect := make(chan struct{})
+					ctx := context.WithValue(cmd.Context(), connectedKey, connect)
+					ctx, cancel := context.WithCancel(ctx)
+					cmd.SetContext(ctx)
+
+					err := make(chan error, 1)
+					go runConnectGoroutine(cmd, err)
+
+					if i == v1.UsbClientPassthroughMaxNumberOf {
+						// Last test is meant to fail.
+						tests = append(tests, session{cancel, err})
+						return true
+					}
+
+					// Till the last test, all sockets must be connected
+					select {
+					case <-connect:
+						tests = append(tests, session{cancel, err})
+						return true
+					case err := <-err:
+						Expect(err).To(MatchError(syscall.ECONNRESET))
```

Comment:
If you want the `Eventually` to repeat and to not fail the test globally it needs to be `g.Expect`.

```suggestion
						g.Expect(err).To(MatchError(syscall.ECONNRESET))
						cancel()
```

Reply:
Thanks!